Jint/Runtime/Interop/MethodDescriptor.cs

```diff
@@ -60,6 +60,17 @@ private static MethodDescriptor[] Prioritize(MethodDescriptor[] descriptors)
         {
             static int CreateComparison(MethodDescriptor d1, MethodDescriptor d2)
             {
+                // if its a generic method, put it on the end
+                if (d1.Method.IsGenericMethod && !d2.Method.IsGenericMethod)
+                {
+                    return 1;
+                }
+
+                if (d2.Method.IsGenericMethod && !d1.Method.IsGenericMethod)
+                {
+                    return -1;
+                }
+
                 // put params versions to end, they can be tricky to match and can cause trouble / extra overhead
                 if (d1.HasParams)
                 {
```

Comment:
Isn't then there the same problem with potentially infinite loops?

Reply:
Probably. This is an issue I found while working on Array.Sort this week. Can you fix it too ?

Reply:
off course, i just wanted to ask before